src/screens/Portfolio/PortfolioHistory.js

```diff
@@ -0,0 +1,172 @@
+import LoadingFooter from "../../components/LoadingFooter";
+import Button from "../../components/Button";
+import { ScreenName } from "../../const";
+
+type Props = {
+  navigation: any,
+};
+
+export function PortfolioHistoryList({
+  onEndReached,
+  opCount = 10,
+  navigation,
+}: {
+  onEndReached?: () => void,
+  opCount?: number,
+  navigation: any,
+}) {
+  const accounts = useSelector(accountsSelector);
+  const allAccounts = useSelector(flattenAccountsSelector);
+
+  const refreshAccountsOrdering = useRefreshAccountsOrdering();
+  useFocusEffect(refreshAccountsOrdering);
+
+  const { sections, completed } = groupAccountsOperationsByDay(accounts, {
+    count: opCount,
+    withSubAccounts: true,
+  });
+
+  function ListEmptyComponent() {
+    if (accounts.length === 0) {
+      return <EmptyStatePortfolio navigation={navigation} />;
+    }
+
+    if (accounts.every(isAccountEmpty)) {
+      return <NoOpStatePortfolio />;
+    }
+
+    return null;
+  }
+
+  function keyExtractor(item: Operation) {
+    return item.id;
+  }
+
+  function renderItem({
+    item,
+    index,
+    section,
+  }: {
+    item: Operation,
+    index: number,
+    section: SectionBase<*>,
+  }) {
+    const account = allAccounts.find(a => a.id === item.accountId);
+    const parentAccount =
+      account && account.type !== "Account"
+        ? accounts.find(a => a.id === account.parentId)
+        : null;
+
+    if (!account) return null;
+
+    return (
+      <OperationRow
+        operation={item}
+        parentAccount={parentAccount}
+        account={account}
+        multipleAccounts
+        isLast={section.data.length - 1 === index}
+      />
+    );
+  }
+
+  function renderSectionHeader({ section }: { section: { day: Date } }) {
+    return <SectionHeader section={section} />;
+  }
+
+  const onTransactionButtonPress = useCallback(() => {
+    navigation.navigate(ScreenName.PortfolioOperationHistory);
+  }, [navigation]);
+
+  return (
+    <SectionList
+      // $FlowFixMe
+      sections={sections}
+      style={styles.list}
+      contentContainerStyle={styles.contentContainer}
+      keyExtractor={keyExtractor}
+      renderItem={renderItem}
+      // $FlowFixMe
+      renderSectionHeader={renderSectionHeader}
+      stickySectionHeadersEnabled={false}
+      onEndReached={onEndReached}
+      ListFooterComponent={
+        !completed ? (
+          !onEndReached ? (
+            <View style={styles.seeMoreBtn}>
+              <Button
+                event="View Transactions"
+                type="lightPrimary"
+                title={<Trans i18nKey="common.seeAll" />}
+                onPress={onTransactionButtonPress}
+              />
+            </View>
+          ) : (
+            <LoadingFooter />
+          )
+        ) : accounts.every(isAccountEmpty) ? null : sections.length ? (
+          <NoMoreOperationFooter />
+        ) : (
+          <NoOperationFooter />
+        )
+      }
```

Comment:
Maybe encapsulate that in a function too, same as `renderSectionHeader` and `ListEmptyComponent`?